Mage.Sets/src/mage/cards/f/Fiendlash.java

```diff
@@ -0,0 +1,155 @@
+
+        // Whenever equipped creature is dealt damage, it deals damage equal to its
+        // power to target player or planeswalker.
+        this.addAbility(new FiendlashTriggeredAbility());
+
+        // Equip {2}{R}
+        this.addAbility(new EquipAbility(Outcome.AddAbility, new ManaCostsImpl<>("{2}{R}")));
+    }
+
+    private Fiendlash(final Fiendlash card) {
+        super(card);
+    }
+
+    @Override
+    public Fiendlash copy() {
+        return new Fiendlash(this);
+    }
+}
+
+class FiendlashTriggeredAbility extends TriggeredAbilityImpl {
+
+    private boolean usedForCombatDamageStep = false;
+
+    FiendlashTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new FiendlashEffect(), false);
+        this.addTarget(new TargetPlayerOrPlaneswalker());
+    }
+
+    private FiendlashTriggeredAbility(final FiendlashTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public FiendlashTriggeredAbility copy() {
+        return new FiendlashTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DAMAGED_PERMANENT
+                || event.getType() == GameEvent.EventType.COMBAT_DAMAGE_STEP_POST;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent equipment = game.getPermanent(this.getSourceId());
+        if (equipment == null || equipment.getAttachedTo() == null) {
+            return false;
+        }
+        if (event.getType() == GameEvent.EventType.DAMAGED_PERMANENT
+                && event.getTargetId().equals(equipment.getAttachedTo())) {
+            this.getEffects().setValue("equipped", game.getPermanent(equipment.getAttachedTo()));
+
+            if (((DamagedEvent) event).isCombatDamage()) {
+                if (!usedForCombatDamageStep) {
+                    usedForCombatDamageStep = true;
+                    return true;
+                }
+            } else {
+                return true;
+            }
+        }
+        if (event.getType() == GameEvent.EventType.COMBAT_DAMAGE_STEP_POST) {
+            usedForCombatDamageStep = false;
+        }
+        return false;
+    }
+
+    @Override
+    public String getRule() {
+        return "Whenever equipped creature is dealt damage, it deals damage equal to its power to target player or planeswalker.";
+    }
+}
+
+class FiendlashEffect extends OneShotEffect {
+
+    FiendlashEffect() {
+        super(Outcome.Benefit);
+    }
+
+    private FiendlashEffect(final FiendlashEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public FiendlashEffect copy() {
+        return new FiendlashEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent creature = (Permanent) getValue("equipped");
+        int damage = (int) creature.getPower().getValue();
+        if (creature == null || damage < 1) {
```

Comment:
Also need the null check before you try to get the creature's power.

Reply:
updated